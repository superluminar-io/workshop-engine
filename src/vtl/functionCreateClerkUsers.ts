export const request = `
  {
    "version": "2018-05-29",
    "operation": "Invoke",
    "payload": {
      "userEmailAddresses": $util.toJson($ctx.arguments.input.attendees)
    }
  }
`;

export const response = `
  $util.toJson($ctx.prev.result)
`;
