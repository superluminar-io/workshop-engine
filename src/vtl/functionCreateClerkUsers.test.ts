import Parser from 'appsync-template-tester';
import * as functionCreateClerkUsers from './functionCreateClerkUsers';

describe('request mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(functionCreateClerkUsers.request);

    const context = {
      arguments: {
        input: {
          attendees: [
            'me@example.com',
          ],
        },
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(functionCreateClerkUsers.response);

    const context = {
      prev: {
        result: {
          any: 'data',
        },
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
