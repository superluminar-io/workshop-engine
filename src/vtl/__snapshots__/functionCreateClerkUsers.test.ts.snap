// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping should match snapshot 1`] = `
"{
    \\"version\\": \\"2018-05-29\\",
    \\"operation\\": \\"Invoke\\",
    \\"payload\\": {
      \\"userEmailAddresses\\": [\\"me@example.com\\"],
    }
  }"
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
